gallery/src/components/NxModal/NxModalFormErrorExample.tsx

```diff
@@ -29,32 +40,21 @@ export default function NxModalFormErrorExample() {
               <div className="nx-form-group">
                 <label className="nx-label">
                   Username
-                  <input type="text"
-                         className="nx-text-input"
-                         placeholder="Username"/>
+                  <NxTextInput { ...state } onChange={onChange} />
                 </label>
               </div>
               <div className="nx-form-group">
                 <label className="nx-label">
                   Password
-                  <input type="password"
-                         className="nx-text-input"
-                         placeholder="Password"/>
+                  <NxTextInput type="password" placeholder="Enter password" onChange={onChange} { ...state }/>
                 </label>
               </div>
             </div>
-            <footer className="nx-modal-footer nx-error">
-              <div id="login-error" className="nx-alert nx-alert--error">
-                <NxFontAwesomeIcon icon={faAngry} />
-                <span>You really messed something up!</span>
-              </div>
+            <footer className="nx-modal-footer">
+              <NxLoadError { ...({ error, retryHandler }) } />
```

Comment:
It seems a little weird to have a load error, as if the form had been saved or something, but then to not also have any sort of Save button, either before or after the retry succeeds.  As it is now, after pressing retry, you're left with only a Cancel button.  Which wouldn't realistically make much sense.

Reply:
> It seems a little weird to have a load error, as if the form had been saved or something, but then to not also have any sort of Save button, either before or after the retry succeeds. As it is now, after pressing retry, you're left with only a Cancel button. Which wouldn't realistically make much sense.

I find it weird that we have a retry button in the alert rather than adjacent to the Cancel button, @dturner32 ?

Reply:
> It seems a little weird to have a load error, as if the form had been saved or something, but then to not also have any sort of Save button, either before or after the retry succeeds. As it is now, after pressing retry, you're left with only a Cancel button. Which wouldn't realistically make much sense.

The Retry button now closes the modal: https://github.com/sonatype/sonatype-react-shared-components/pull/75/commits/d0b9587a5d0daaee57666433d2e46ac006e37919

Reply:
> > It seems a little weird to have a load error, as if the form had been saved or something, but then to not also have any sort of Save button, either before or after the retry succeeds. As it is now, after pressing retry, you're left with only a Cancel button. Which wouldn't realistically make much sense.
> 
> I find it weird that we have a retry button in the alert rather than adjacent to the Cancel button, @dturner32 ?

I included the retry button in the error alert because I was under the intention that that was the desired action a user would take in this case. I agree that it makes more sense to include the x/close instead. If a username or password is entered incorrectly the alert message would reflect that and the user could go back and retry be re-entering their credentials. To Ross' point I think a save or submit button should be included as well.